django/db/migrations/operations/models.py

```diff
@@ -716,3 +716,79 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
 
     def describe(self):
         return "Change managers on %s" % (self.name, )
+
+
+class IndexOperation(Operation):
+
+    def __init__(self, model_name, index):
+        self.model_name = model_name
+        self.index = index
+
+    def deconstruct(self):
+        kwargs = {
+            'model_name': self.model_name,
+            'index': self.index,
+        }
+        return (
+            self.__class__.__name__,
+            [],
+            kwargs,
+        )
+
+
+class CreateIndex(IndexOperation):
+    """
+    Creates an index for a model.
+    """
+
+    def state_forwards(self, app_label, state):
+        model_state = state.models[app_label, self.model_name.lower()]
+        if not hasattr(self.index, 'model'):
+            self.index.model = state.apps.get_model(app_label, self.model_name)
+        if 'indexes' not in model_state.options:
+            model_state.options['indexes'] = []
+        model_state.options['indexes'].append(self.index)
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        schema_editor.create_index(
+            self.index,
+        )
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        schema_editor.drop_index(
+            self.index,
+        )
+
+    def describe(self):
+        return "Create index on field(s) %s of model %s" % (
```

Comment:
The index name is not generated with the initialization of an instance of `Index` (bcoz the current implementation in Django makes it database backend dependent) unlike the describe method which should be available as soon as an instance is created.